clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -307,13 +333,41 @@ public long timeToNextPoll(long now) {
         client.ensureFreshMetadata();
 
         isLeader = true;
-        assignmentSnapshot = metadataSnapshot;
 
         log.debug("Performing assignment for group {} using strategy {} with subscriptions {}",
                 groupId, assignor.name(), subscriptions);
 
         Map<String, Assignment> assignment = assignor.assign(metadata.fetch(), subscriptions);
 
+        // user-customized assignor may have created some topics that are not in the subscription list
```

Comment:
Can we mention in this comment that this is a hack and not something we want to support long-term unless we push regex into the protocol?